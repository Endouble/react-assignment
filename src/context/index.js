import React from 'react';

export const defaultCards = [];
export const defaultFilter = {
    value: '',
    filterBy: 'name',
};
export const defaultIsLoading = false;

export const CardsContext = React.createContext(defaultCards);
export const FilterContext = React.createContext(defaultFilter);
export const IsLoadingContext = React.createContext(defaultIsLoading);
