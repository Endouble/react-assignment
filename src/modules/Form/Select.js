import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { formElement } from '../../styles/mixins';

class Select extends React.Component {
    constructor(props) {
        super(props);
        this.handleChange = this.handleChange.bind(this);
    }

    handleChange(event) {
        const { value } = this.props.filter;
        const filterBy = event.target.value;
        this.props.handleChange({
            value,
            filterBy,
        });
    }

    render() {
        const { options } = this.props;
        return (
            <StyledSelect onChange={this.handleChange}>
                {options.map(option => <option key={`${option}-opt`} value={option}>{option}</option>)}
            </StyledSelect>
        );
    }
}

/*
    Select Styles
*/
const StyledSelect = styled.select`
    ${formElement()};
`;

/*
    Select propTypes
*/
Select.propTypes = {
    handleChange: PropTypes.func,
    filter: PropTypes.shape({
        value: PropTypes.string,
    }),
    options: PropTypes.arrayOf(PropTypes.string),
};

Select.defaultProps = {
    handleChange: null,
    filter: {},
    options: [],
};


export default Select;
