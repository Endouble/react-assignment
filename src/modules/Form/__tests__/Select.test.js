import React from 'react';
import 'jsdom-global/register';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Select from '../Select';

import { availableFilters } from '../../../context';

configure({ adapter: new Adapter() });

describe('Select', () => {
    it('renders correctly', () => {
        const component = shallow(<Select />);
        expect(component).toMatchSnapshot();
    });

    it('Has option', () => {
        const component = mount(<Select options={availableFilters} />);
        const option = component.find('option').first();
        expect(option.prop('value')).toEqual(availableFilters[0]);
    });
});
