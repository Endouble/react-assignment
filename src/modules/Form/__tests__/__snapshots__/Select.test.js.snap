// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly 1`] = `
<styled.select
  onChange={[Function]}
>
  <option
    key="name-opt"
    value="name"
  >
    name
  </option>
  <option
    key="set-opt"
    value="set"
  >
    set
  </option>
  <option
    key="artist-opt"
    value="artist"
  >
    artist
  </option>
</styled.select>
`;
